File: leetcode/Plus_One.java
import java.util.*;

public class Solution {
    public int[] plusOne(int[] digits) {
        int a = 1;
        for (int i = digits.length - 1; i >= 0; --i) {
            int s = digits[i] + a;
            digits[i] = s % 10;
            a = s / 10;
        }
        if (a == 0) return digits;
        int[] res = new int[digits.length + 1];
        res[0] = a;
        System.arraycopy(digits, 0, res, 1, digits.length);
        return res;
    }
}

File: leetcode/Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        vector<int> after_addition(digits.size());
        int inc = 1;
        for (int i = digits.size() - 1; i >= 0; --i) {
            int sum = digits[i] + inc;
            after_addition[i] = sum % 10;
            inc = sum / 10;
        }
        if (inc == 1) {
            after_addition[0] = 1;
            after_addition.push_back(0);
        }
        return after_addition;
    }
};
